OutOfSchool/OutOfSchool.WebApi/Services/Database/WorkshopService.cs

```diff
@@ -387,6 +387,24 @@ public async Task<IEnumerable<Workshop>> PartialUpdateByProvider(Provider provid
         }
     }
 
+    /// <inheritdoc/>
+    /// <exception cref="DbUpdateConcurrencyException">If a concurrency violation is encountered while saving to database.</exception>
+    public async Task<IEnumerable<Workshop>> BlockByProvider(Provider provider)
+    {
+        logger.LogInformation($"Block {nameof(Workshop)} with ProviderId = {provider?.Id} was started.");
```

Comment:
As I see provider can't be null here because checking for null is happened at a more high level, in the method public async Task<ProviderBlockDto> **Block**(ProviderBlockDto providerBlockDto) in **ProviderSerice.cs.**

So your suggestion is to add check-in on null here too? And throw ArgumentNullException?

Reply:
We may perform all argument validation check in all public methods, and 'trust' to everything private/internal